avocado/core/tree.py

```diff
@@ -107,6 +107,12 @@ def __ne__(self, other):
         """ Inverted eq """
         return not self == other
 
+    def fingerprint(self):
+        """
+        Reports string which represents the value of this node.
+        """
+        return str(self.path) + str(self.environment) + str(self.ctrl)
```

Comment:
sure, btw the speed of formatting vs. adding is not 100% like that, it depends on many factors, but I can use it that way. And yes, the `%s` means convert it to string, the conversion is only necessary when using the `+`.